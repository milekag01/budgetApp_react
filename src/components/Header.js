import React from 'react';
import { NavLink } from 'react-router-dom';

const Header= () => (
    <div>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Go Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add new Expense</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink>
        <NavLink to="/help" activeClassName="is-active">Get Help</NavLink>
    </div>
);

export default Header;