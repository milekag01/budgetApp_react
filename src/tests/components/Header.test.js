import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json';         //since we are using enzyme-to-json/serializer, we do not need to import and use (enzyme-to-json) since it will be used automatically
import Header from '../../components/Header';

test('should render header correctly',() => {
    const wrapper = shallow(<Header />);
    // expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();

    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

