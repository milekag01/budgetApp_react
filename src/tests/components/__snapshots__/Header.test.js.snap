// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Go Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    Add new Expense
  </NavLink>
</div>
`;
